sessionctx/binloginfo/binloginfo.go

```diff
@@ -150,17 +153,37 @@ func SetDDLBinlog(client *pumpcli.PumpsClient, txn kv.Transaction, jobID int64,
 
 const specialPrefix = `/*!90000 `
 
-func addSpecialComment(ddlQuery string) string {
+// AddSpecialComment uses to add comment for table option in DDL query.
+// Export for testing.
+func AddSpecialComment(ddlQuery string) string {
 	if strings.Contains(ddlQuery, specialPrefix) {
 		return ddlQuery
 	}
-	loc := shardPat.FindStringIndex(strings.ToUpper(ddlQuery))
-	if loc == nil {
+	ddlQuery = addSpecialCommentByRegexp(ddlQuery, shardPat)
```

Comment:
Rather than adding special comments, then combining them by extracting in regexp pattern, how about firstly storing all them in some struct, say, a `slice` and then generating the final comment at once?

The latter approach has 2 advantages:

1. It has better performance with less regexp matching.
2. If we have more special comments in the future, they will be easy to handle